packages/@aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts

```diff
@@ -16,6 +16,32 @@ export type AwsSdkMetadata = {[key: string]: any};
 
 const awsSdkMetadata: AwsSdkMetadata = metadata;
 
+/**
+ * Physical ID of the custom resource.
+ */
+export class PhysicalResourceId {
+
+  /**
+   * Construct a resource id from the path to the data in the API call response.
+   */
+  public static fromResponsePath(responsePath: string): PhysicalResourceId {
+    return new PhysicalResourceId(responsePath, undefined);
+  }
+
+  /**
+   * Construct a resource id from an explicit id.
+   */
+  public static fromLiteralString(literalId: string): PhysicalResourceId {
```

Comment:
Sometimes we use `of` to denote a literal value: `PhysicalResourceId.of(xxx)`.

Reply:
Nice, I was indeed not so comfortable with the name 👍